crates/agent/src/resource_configs.rs

```diff
@@ -1,34 +1,79 @@
+use models::{SourceCaptureSchemaMode, SourceCapture};
 use serde_json::Value;
 
 ///
 /// # Panics
 /// If the `full_collection_name` doesn't contain any `/` characters, which should never
 /// be the case since we should have already validated the collection name.
 pub fn update_materialization_resource_spec(
+    source_capture: &SourceCapture,
     resource_spec: &mut Value,
-    collection_name_ptr: &doc::Pointer,
+    resource_spec_pointers: &ResourceSpecPointers,
     full_collection_name: &str,
-) -> anyhow::Result<Value> {
-    let resource_name = full_collection_name
-        .rsplit_once('/')
-        .expect("collection name is invalid (does not contain '/')")
-        .1
-        .to_owned();
-
-    let Some(prev) = collection_name_ptr.create_value(resource_spec) else {
+) -> anyhow::Result<()> {
+    let split: Vec<&str> = full_collection_name
+        .rsplit('/')
+        .take(2)
+        .collect();
+
+    if split.len() < 2 {
+        return Err(anyhow::anyhow!("collection name is invalid (does not contain '/')"))
+    }
+
+    let x_collection_name = split[0];
+    let x_schema_name = split[1];
+
+    let x_collection_name_ptr = &resource_spec_pointers.x_collection_name;
+
+    let Some(x_collection_name_prev) = x_collection_name_ptr.create_value(resource_spec) else {
         anyhow::bail!(
-            "cannot create location '{collection_name_ptr}' in resource spec '{resource_spec}'"
+            "cannot create location '{x_collection_name_ptr}' in resource spec '{resource_spec}'"
         );
     };
 
-    Ok(std::mem::replace(prev, resource_name.into()))
+    let _ = std::mem::replace(x_collection_name_prev, x_collection_name.into());
+
+    let source_capture_def = source_capture.to_normalized_def();
+
+    if source_capture_def.target_schema == SourceCaptureSchemaMode::FromSourceName {
+        if let Some(x_schema_name_ptr) = &resource_spec_pointers.x_schema_name {
+            if let Some(x_schema_name_prev) = x_schema_name_ptr.create_value(resource_spec) {
+                let _ = std::mem::replace(x_schema_name_prev, x_schema_name.into());
+            } else {
+                anyhow::bail!(
+                    "cannot create location '{x_schema_name_ptr}' in resource spec '{resource_spec}'"
+                );
+            }
+        }
+    }
+
+    if source_capture_def.delta_updates {
+        if let Some(x_delta_updates_ptr) = &resource_spec_pointers.x_delta_updates {
```

Comment:
Similar to `target_schema`, it seems like we should validate that the source capture configuration is consistent with the resource spec schema of the connector, or at least log something if it's not.